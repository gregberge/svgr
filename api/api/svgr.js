/* eslint-disable import/no-unresolved */
/* eslint-disable import/no-extraneous-dependencies */
const { default: svgr } = require('@svgr/core')
const { default: jsx } = require('@svgr/plugin-jsx')
const { default: svgo } = require('@svgr/plugin-svgo')
const { default: prettier } = require('@svgr/plugin-prettier')

module.exports = (req, res) => {
  if (!req.body) {
    res.status(204).send('')
    return
  }
  svgr(req.body.code, { ...req.body.options, plugins: [svgo, jsx, prettier] })
    .then((output) => {
      res.status(200).json({ output })
    })
    .catch((error) => {
      res.status(400).json({ error: error.message })
    })
}
