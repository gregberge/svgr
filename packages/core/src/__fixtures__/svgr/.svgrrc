{
    noSemi: true,
    icon: true,
    replaceAttrValues: [["#063855", "currentColor"]]
}