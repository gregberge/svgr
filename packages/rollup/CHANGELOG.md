# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [4.3.0](https://github.com/smooth-code/svgr/tree/master/packages/rollup/compare/v4.2.0...v4.3.0) (2019-05-28)

**Note:** Version bump only for package @svgr/rollup





# [4.2.0](https://github.com/smooth-code/svgr/tree/master/packages/rollup/compare/v4.1.0...v4.2.0) (2019-04-11)

**Note:** Version bump only for package @svgr/rollup





# [4.1.0](https://github.com/smooth-code/svgr/compare/v4.0.4...v4.1.0) (2018-11-24)


### Features

* add parcel plugin ([#235](https://github.com/smooth-code/svgr/issues/235)) ([144dbe3](https://github.com/smooth-code/svgr/commit/144dbe3)), closes [#215](https://github.com/smooth-code/svgr/issues/215)





## [4.0.4](https://github.com/smooth-code/svgr/compare/v4.0.3...v4.0.4) (2018-11-24)


### Bug Fixes

* **webpack:** use static babel config ([#240](https://github.com/smooth-code/svgr/issues/240)) ([d67af31](https://github.com/smooth-code/svgr/commit/d67af31)), closes [#232](https://github.com/smooth-code/svgr/issues/232)





## [4.0.3](https://github.com/smooth-code/svgr/compare/v4.0.2...v4.0.3) (2018-11-13)


### Bug Fixes

* upgrade dependencies ([7e2195f](https://github.com/smooth-code/svgr/commit/7e2195f))





## [4.0.2](https://github.com/smooth-code/svgr/compare/v4.0.1...v4.0.2) (2018-11-08)

**Note:** Version bump only for package @svgr/rollup





## [4.0.1](https://github.com/smooth-code/svgr/compare/v4.0.0...v4.0.1) (2018-11-08)

**Note:** Version bump only for package @svgr/rollup





# [4.0.0](https://github.com/smooth-code/svgr/compare/v3.1.0...v4.0.0) (2018-11-04)


### Features

* **v4:** new architecture ([ac8b8ca](https://github.com/smooth-code/svgr/commit/ac8b8ca))


### BREAKING CHANGES

* **v4:** - `template` option must now returns a Babel AST
- `@svgr/core` does not include svgo & prettier by default





# [3.1.0](https://github.com/smooth-code/svgr/compare/v3.0.0...v3.1.0) (2018-10-05)


### Bug Fixes

* style & custom SVG properties ([#203](https://github.com/smooth-code/svgr/issues/203)) ([f8b2212](https://github.com/smooth-code/svgr/commit/f8b2212)), closes [#199](https://github.com/smooth-code/svgr/issues/199) [#201](https://github.com/smooth-code/svgr/issues/201)





<a name="3.0.0"></a>
# [3.0.0](https://github.com/smooth-code/svgr/compare/v2.4.1...v3.0.0) (2018-10-01)


### Bug Fixes

* **rollup:** forward filePath in rollup plugin ([461492b](https://github.com/smooth-code/svgr/commit/461492b)), closes [#177](https://github.com/smooth-code/svgr/issues/177) [#188](https://github.com/smooth-code/svgr/issues/188)


### Features

* always prefix component name with "Svg" ([f71aa7a](https://github.com/smooth-code/svgr/commit/f71aa7a)), closes [#190](https://github.com/smooth-code/svgr/issues/190)


### BREAKING CHANGES

* **rollup:** runtime configuration is now loaded using rollup plugin.





<a name="2.4.1"></a>
## [2.4.1](https://github.com/smooth-code/svgr/compare/v2.4.0...v2.4.1) (2018-09-16)

**Note:** Version bump only for package @svgr/rollup





<a name="2.4.0"></a>
# [2.4.0](https://github.com/smooth-code/svgr/compare/v2.3.0...v2.4.0) (2018-09-16)


### Features

* **upgrade:** h2x@1.1.0 (jsdom@12.0.0) & others ([2d9b7bd](https://github.com/smooth-code/svgr/commit/2d9b7bd))





<a name="2.3.0"></a>
# [2.3.0](https://github.com/smooth-code/svgr/compare/v2.2.1...v2.3.0) (2018-09-03)


### Features

* upgrade to Babel v7 ([7bc908d](https://github.com/smooth-code/svgr/commit/7bc908d))





<a name="2.2.1"></a>
## [2.2.1](https://github.com/smooth-code/svgr/compare/v2.2.0...v2.2.1) (2018-08-16)


### Bug Fixes

* **rollup:** fix to work with rollup-plugin-typescript2 ([#147](https://github.com/smooth-code/svgr/issues/147)) ([4b3737e](https://github.com/smooth-code/svgr/commit/4b3737e))





<a name="2.2.0"></a>
# [2.2.0](https://github.com/smooth-code/svgr/compare/v2.1.1...v2.2.0) (2018-08-13)

**Note:** Version bump only for package @svgr/rollup





<a name="2.1.1"></a>
## [2.1.1](https://github.com/smooth-code/svgr/compare/v2.1.0...v2.1.1) (2018-07-11)




**Note:** Version bump only for package @svgr/rollup

<a name="2.1.0"></a>
# [2.1.0](https://github.com/smooth-code/svgr/compare/v2.0.0...v2.1.0) (2018-07-08)


### Features

* **cli:** support custom filename cases ([#136](https://github.com/smooth-code/svgr/issues/136)) ([4922f7a](https://github.com/smooth-code/svgr/commit/4922f7a)), closes [#118](https://github.com/smooth-code/svgr/issues/118)
