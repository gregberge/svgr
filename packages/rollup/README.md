# @svgr/rollup

[![Build Status](https://img.shields.io/travis/smooth-code/svgr.svg)](https://travis-ci.org/smooth-code/svgr)
[![Version](https://img.shields.io/npm/v/@svgr/rollup.svg)](https://www.npmjs.com/package/@svgr/rollup)
[![MIT License](https://img.shields.io/npm/l/@svgr/rollup.svg)](https://github.com/smooth-code/svgr/blob/master/LICENSE)

Rollup plugin for SVGR.

```
npm install @svgr/rollup --save-dev
```

In your `rollup.config.js`:

```js
{
  plugins: [svgr()]
}
```

In your code:

```js
import Star from './star.svg'

const App = () => (
  <div>
    <Star />
  </div>
)
```

### Passing options

```js
{
  plugins: [svgr({ native: true })]
}
```

### Using with `url` plugin

It is possible to use it with [`url`](https://github.com/rollup/rollup-plugin-url).

In your `rollup.config.js`:

```js
{
  plugins: [url(), svgr()]
}
```

In your code:

```js
import starUrl, { ReactComponent as Star } from './star.svg'

const App = () => (
  <div>
    <img src={starUrl} alt="star" />
    <Star />
  </div>
)
```

The named export defaults to `ReactComponent`, but can be customized with the `namedExport` option.

Please note that by default, `@svgr/rollup` will try to export the React Component via default export if there is no other plugin handling svg files with default export. When there is already any other plugin using default export for svg files, `@svgr/rollup` will always export the React component via named export.

If you prefer named export in any case, you may set the `exportType` option to `named`.

### Use your own Babel configuration

By default, `@svgr/rollup` applies a babel transformation with [optimized configuration](https://github.com/gregberge/svgr/blob/main/packages/rollup/src/index.ts). In some case you may want to apply a custom one (if you are using Preact for an example). You can turn off Babel transformation by specifying `babel: false` in options.

```js
{
  plugins: [svgr({ babel: false })]
}
```

## License

MIT
