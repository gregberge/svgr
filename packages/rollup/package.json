{
  "name": "@svgr/rollup",
  "description": "SVGR Rollup plugin.",
  "version": "2.4.1",
  "main": "lib/index.js",
  "repository": "git@github.com:smooth-code/svgr.git",
  "author": "Greg Berg√© <berge.greg@gmail.com>",
  "keywords": [
    "svgr",
    "svg",
    "react",
    "rollup",
    "rollup-plugin"
  ],
  "engines": {
    "node": ">=8"
  },
  "license": "MIT",
  "scripts": {
    "prebuild": "rm -rf lib/",
    "build": "babel --config-file ../../babel.config.js -d lib --ignore \"**/*.test.js\" src",
    "prepublishOnly": "yarn run build"
  },
  "dependencies": {
    "@babel/core": "^7.1.2",
    "@babel/plugin-transform-react-constant-elements": "^7.0.0",
    "@babel/preset-env": "^7.1.0",
    "@babel/preset-react": "^7.0.0",
    "@svgr/core": "^2.4.1",
    "rollup-pluginutils": "^2.3.3"
  },
  "devDependencies": {
    "rollup": "^0.66.2",
    "rollup-plugin-image": "^1.0.2",
    "rollup-plugin-url": "^1.4.0"
  }
}
