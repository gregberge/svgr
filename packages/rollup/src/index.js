import fs from 'fs'
import convert from '@svgr/core'
import { createFilter } from 'rollup-pluginutils'
import { transform as babelTransform } from '@babel/core'

function svgrPlugin(options = {}) {
  const filter = createFilter(options.include || '**/*.svg', options.exclude)
  const { babel = true } = options

  return {
    async transform(data, id) {
      if (!filter(id)) return null
      if (id.slice(-4) !== '.svg') return null

      const load = fs.readFileSync(id, 'utf8')

      const exportMatches =
        data.match(/^module.exports\s*=\s*(.*)/) ||
        data.match(/export\sdefault\s(.*)/)

      const previousExport = exportMatches ? data : null

      const ast = {
        type: 'Program',
        sourceType: 'module',
        start: 0,
        end: null,
        body: [],
      }

      const jsCode = await convert(load, options, {
        rollup: { previousExport },
      })

      const pBabelTransform = async code =>
        new Promise((resolve, reject) => {
          babelTransform(
            code,
            {
              babelrc: false,
              presets: [
                '@babel/preset-react',
                ['@babel/preset-env', { modules: false }],
              ],
              plugins: [
                '@babel/plugin-transform-react-constant-elements',
                '@babel/plugin-proposal-object-rest-spread',
              ],
            },
            (err, result) => {
              if (err) reject(err)
              else resolve(result.code)
            },
          )
        })

      const code = babel ? await pBabelTransform(jsCode) : jsCode

      return { ast, code, map: { mappings: '' } }
    },
  }
}

export default svgrPlugin
