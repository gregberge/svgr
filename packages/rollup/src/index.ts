import * as fs from 'fs'
import { transform, Config } from '@svgr/core'
import { createFilter, CreateFilter } from '@rollup/pluginutils'
import { transformAsync, createConfigItem } from '@babel/core'
import svgo from '@svgr/plugin-svgo'
import jsx from '@svgr/plugin-jsx'
// @ts-ignore
import presetReact from '@babel/preset-react'
// @ts-ignore
import presetEnv from '@babel/preset-env'
// @ts-ignore
import presetTS from '@babel/preset-typescript'
// @ts-ignore
import pluginTransformReactConstantElements from '@babel/plugin-transform-react-constant-elements'
import type { PluginImpl } from 'rollup'

const babelOptions = {
  babelrc: false,
  configFile: false,
  presets: [
    createConfigItem(presetReact, { type: 'preset' }),
    createConfigItem([presetEnv, { modules: false }], { type: 'preset' }),
  ],
  plugins: [createConfigItem(pluginTransformReactConstantElements)],
}

const typeScriptBabelOptions = {
  ...babelOptions,
  presets: [
    ...babelOptions.presets,
    createConfigItem(
      [presetTS, { allowNamespaces: true, allExtensions: true, isTSX: true }],
      { type: 'preset' },
    ),
  ],
}
export interface Options extends Config {
  include?: Parameters<CreateFilter>[0]
  exclude?: Parameters<CreateFilter>[1]
  babel?: boolean
}

const plugin: PluginImpl<Options> = (options = {}) => {
  const EXPORT_REGEX = /(module\.exports *= *|export default)/
  const filter = createFilter(options.include || '**/*.svg', options.exclude)
  const { babel = true } = options

  return {
    name: 'svgr',
    async transform(data, id) {
      if (!filter(id)) return null
      if (id.slice(-4) !== '.svg') return null

      const load = fs.readFileSync(id, 'utf8')

      const previousExport = EXPORT_REGEX.test(data) ? data : null

      const jsCode = await transform(load, options, {
        filePath: id,
        caller: {
          name: '@svgr/rollup',
          previousExport,
          defaultPlugins: [svgo, jsx],
        },
      })

      if (babel) {
        const result = await transformAsync(
          jsCode,
          options.typescript ? typeScriptBabelOptions : babelOptions,
        )
        if (!result?.code) {
          throw new Error(`Error while transforming using Babel`)
        }
        return { code: result.code, map: null }
      }

      return {
        ast: {
          type: 'Program',
          start: 0,
          end: 0,
          sourceType: 'module',
          body: [],
        },
        code: jsCode,
        map: null
      }
    },
  }
}

export default plugin
