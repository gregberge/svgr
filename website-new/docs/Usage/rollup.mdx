---
section: Usage
title: Rollup.js
slug: /docs/rollup/
sidebar_position: 50
---

# Rollup.js

SVGR provides an official [rollup.js](https://rollupjs.org) plugin to import SVG as React components.

<carbon-ad />

## Install

```bash
npm install --save-dev @svgr/rollup
# or use yarn
yarn add --dev @svgr/rollup
```

## Usage

**rollup.config.js**

```js
import svgr from '@svgr/rollup'

export default {
  plugins: [svgr()],
  input: 'src/main.js',
  output: {
    file: 'bundle.js',
    format: 'cjs',
  },
}
```

**Your code**

```js
import Star from './star.svg'

const App = () => (
  <div>
    <Star />
  </div>
)
```

## Options

SVGR let you specify options in a runtime config file like `svgr.config.js` or directly in the rollup.js plugin:

**rollup.config.js**

```js
import svgr from '@svgr/rollup'

export default {
  plugins: [svgr({ icon: true })],
  input: 'src/main.js',
  output: {
    file: 'bundle.js',
    format: 'cjs',
  },
}
```

[SVGR options reference](https://react-svgr.com/docs/options/) describes all options available.

## Using with `@rollup/plugin-url`

It is possible to use it with [`@rollup/plugin-url`](https://github.com/rollup/plugins/tree/master/packages/url).

**rollup.config.js**

```js
import url from '@rollup/plugin-url'
import svgr from '@svgr/rollup'

export default {
  plugins: [url(), svgr({ icon: true })],
  input: 'src/main.js',
  output: {
    file: 'bundle.js',
    format: 'cjs',
  },
}
```

**Your code**

```js
import starUrl, { ReactComponent as Star } from './star.svg'

const App = () => (
  <div>
    <img src={starUrl} alt="star" />
    <Star />
  </div>
)
```

The named export defaults to `ReactComponent`, but can be customized with the `namedExport` option.

Please note that by default, `@svgr/rollup` will try to export the React Component via default export if there is no other plugin handling svg files with default export. When there is already any other plugin using default export for svg files, `@svgr/rollup` will always export the React component via named export.

If you prefer named export in any case, you may set the `exportType` option to `named`.

### Use your own Babel configuration

By default, `@svgr/rollup` applies a babel transformation with [optimized configuration](https://github.com/gregberge/svgr/blob/main/packages/rollup/src/index.ts). In some case you may want to apply a custom one (if you are using Preact for an example). You can turn off Babel transformation by specifying `babel: false` in options.

```js
{
  plugins: [svgr({ babel: false })]
}
```
