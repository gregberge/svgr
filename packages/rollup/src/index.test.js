import { rollup } from 'rollup'
import image from 'rollup-plugin-image'
import url from 'rollup-plugin-url'
import svgr from './index'

const compile = (plugins = [svgr()]) =>
  rollup({
    input: './__fixtures__/simple/file.svg',
    plugins,
  })

const getCode = bundler =>
  bundler.modules.find(({ id }) => id.includes('__fixtures__/simple/file.svg') || id.includes('__fixtures__\\simple\\file.svg'))
    .code

describe('rollup loader', () => {
  it('should convert file', async () => {
    const code = await compile([svgr()])
    expect(getCode(code)).toMatchSnapshot()
  })

  it('should convert file without babel', async () => {
    const code = await compile([svgr({ babel: false })])
    expect(getCode(code)).toMatchSnapshot()
  })

  it('should convert file with previousExport of image plugin', async () => {
    const code = await compile([image(), svgr({ babel: false })])
    expect(getCode(code)).toMatchSnapshot()
  })

  it('should convert file with previousExport of url plugin', async () => {
    const code = await compile([url(), svgr({ babel: false })])
    expect(getCode(code)).toMatchSnapshot()
  })
})
