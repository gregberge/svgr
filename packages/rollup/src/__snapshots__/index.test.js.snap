// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup loader should convert file 1`] = `
"function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';

var _ref =
/*#__PURE__*/
React.createElement(\\"path\\", {
  d: \\"M0 0h48v1H0z\\",
  fill: \\"#063855\\",
  fillRule: \\"evenodd\\"
});

var SvgComponent = function SvgComponent(props) {
  return React.createElement(\\"svg\\", _extends({
    width: 48,
    height: 1
  }, props), _ref);
};

export default SvgComponent;"
`;

exports[`rollup loader should convert file with previousExport of image plugin 1`] = `
"import React from 'react'

const SvgComponent = props => (
  <svg width={48} height={1} {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

var img = new Image()
img.src =
  'data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iNDhweCIgaGVpZ2h0PSIxcHgiIHZpZXdCb3g9IjAgMCA0OCAxIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPgogICAgPCEtLSBHZW5lcmF0b3I6IFNrZXRjaCA0Ni4yICg0NDQ5NikgLSBodHRwOi8vd3d3LmJvaGVtaWFuY29kaW5nLmNvbS9za2V0Y2ggLS0+CiAgICA8dGl0bGU+UmVjdGFuZ2xlIDU8L3RpdGxlPgogICAgPGRlc2M+Q3JlYXRlZCB3aXRoIFNrZXRjaC48L2Rlc2M+CiAgICA8ZGVmcz48L2RlZnM+CiAgICA8ZyBpZD0iUGFnZS0xIiBzdHJva2U9Im5vbmUiIHN0cm9rZS13aWR0aD0iMSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgICAgICA8ZyBpZD0iMTktU2VwYXJhdG9yIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTI5LjAwMDAwMCwgLTE1Ni4wMDAwMDApIiBmaWxsPSIjMDYzODU1Ij4KICAgICAgICAgICAgPGcgaWQ9IkNvbnRyb2xzL1NldHRpbmdzIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSg4MC4wMDAwMDAsIDAuMDAwMDAwKSI+CiAgICAgICAgICAgICAgICA8ZyBpZD0iQ29udGVudCIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMC4wMDAwMDAsIDY0LjAwMDAwMCkiPgogICAgICAgICAgICAgICAgICAgIDxnIGlkPSJHcm91cCIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMjQuMDAwMDAwLCA1Ni4wMDAwMDApIj4KICAgICAgICAgICAgICAgICAgICAgICAgPGcgaWQ9Ikdyb3VwLTIiPgogICAgICAgICAgICAgICAgICAgICAgICAgICAgPHJlY3QgaWQ9IlJlY3RhbmdsZS01IiB4PSIyNSIgeT0iMzYiIHdpZHRoPSI0OCIgaGVpZ2h0PSIxIj48L3JlY3Q+CiAgICAgICAgICAgICAgICAgICAgICAgIDwvZz4KICAgICAgICAgICAgICAgICAgICA8L2c+CiAgICAgICAgICAgICAgICA8L2c+CiAgICAgICAgICAgIDwvZz4KICAgICAgICA8L2c+CiAgICA8L2c+Cjwvc3ZnPg=='
export default img
export { SvgComponent as ReactComponent }
"
`;

exports[`rollup loader should convert file with previousExport of url plugin 1`] = `
"import React from 'react'

const SvgComponent = props => (
  <svg width={48} height={1} {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default 'data:image/svg+xml,%3C%3Fxml%20version%3D%221.0%22%20encoding%3D%22UTF-8%22%3F%3E%3Csvg%20width%3D%2248px%22%20height%3D%221px%22%20viewBox%3D%220%200%2048%201%22%20version%3D%221.1%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%3E%20%20%20%20%20%20%20%20%3Ctitle%3ERectangle%205%3C%2Ftitle%3E%20%20%20%20%3Cdesc%3ECreated%20with%20Sketch.%3C%2Fdesc%3E%20%20%20%20%3Cdefs%3E%3C%2Fdefs%3E%20%20%20%20%3Cg%20id%3D%22Page-1%22%20stroke%3D%22none%22%20stroke-width%3D%221%22%20fill%3D%22none%22%20fill-rule%3D%22evenodd%22%3E%20%20%20%20%20%20%20%20%3Cg%20id%3D%2219-Separator%22%20transform%3D%22translate%28-129.000000%2C%20-156.000000%29%22%20fill%3D%22%23063855%22%3E%20%20%20%20%20%20%20%20%20%20%20%20%3Cg%20id%3D%22Controls%2FSettings%22%20transform%3D%22translate%2880.000000%2C%200.000000%29%22%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Cg%20id%3D%22Content%22%20transform%3D%22translate%280.000000%2C%2064.000000%29%22%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Cg%20id%3D%22Group%22%20transform%3D%22translate%2824.000000%2C%2056.000000%29%22%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Cg%20id%3D%22Group-2%22%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3Crect%20id%3D%22Rectangle-5%22%20x%3D%2225%22%20y%3D%2236%22%20width%3D%2248%22%20height%3D%221%22%3E%3C%2Frect%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3C%2Fg%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3C%2Fg%3E%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%3C%2Fg%3E%20%20%20%20%20%20%20%20%20%20%20%20%3C%2Fg%3E%20%20%20%20%20%20%20%20%3C%2Fg%3E%20%20%20%20%3C%2Fg%3E%3C%2Fsvg%3E'
export { SvgComponent as ReactComponent }
"
`;

exports[`rollup loader should convert file without babel 1`] = `
"import React from 'react'

const SvgComponent = props => (
  <svg width={48} height={1} {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgComponent
"
`;
